### PR TITLE
Features/navbar skeleton 73

### PR DESCRIPTION
**Pull request by djarredon**
Added in a navigation bar including empty links to all planned pages:
- Sign in
- Sign out
- Profile view
- Edit profile
- Admin view
- About page

### CODE REVIEW

File: client/src/containers/home/index.js [hunk 1]
```diff
@@ -3,21 +3,24 @@ import { push } from 'react-router-redux'
 import { bindActionCreators } from 'redux'
 import { connect } from 'react-redux'
 
+
```
Comment (iandusenbury) [inline client/src/containers/home/index.js]: remove line

File: client/src/containers/app/index.js [hunk 2]
```diff
@@ -19,20 +21,38 @@ class App extends Component {
...
+/*
+
+        <header>
+          <Link to="/">Home</Link>
+          <Link to="/about-us">About</Link>
+		  <Link to="/navbar">Navbar</Link>
+		  <Link to="/demo">Demo</Link>
+        </header>
+
+        <main>
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: delete this comment block

File: client/src/containers/app/index.js [hunk 3]
```diff
@@ -19,20 +21,38 @@ class App extends Component {
...
-        <header>
-          <Link to="/">Home</Link>
-          <Link to="/about-us">About</Link>
-        </header>
+		<Navbar />
 
         <main>
           <Route exact path="/" component={Home} />
           <Route exact path="/about-us" component={About} />
+          <Route exact path="/navbar" component={Navbar} />
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: Remove line, we don't need a route for the navbar

File: client/src/containers/app/index.js [hunk 4]
```diff
@@ -19,20 +21,38 @@ class App extends Component {
   render() {
     return (
       <div>
-        <header>
-          <Link to="/">Home</Link>
-          <Link to="/about-us">About</Link>
-        </header>
+		<Navbar />
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: indentation

File: client/src/containers/app/index.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import React, { Component } from 'react';
-import { Route, Link } from 'react-router-dom'
+import { Route } from 'react-router-dom'
 import Home from '../home'
 import About from '../about'
+import Navbar from '../navbar'
+import Demo from '../demo'
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove line

File: client/package.json [hunk 6]
```diff
@@ -21,6 +26,7 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test --env=jsdom",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "toolbox": "react-toolbox-themr"
```
Comment (iandusenbury) [inline client/package.json]: remove anything eject and react-toolbox related
afterwards be sure to run 'yarn' on the command line to update the yarn.lock file

File: client/src/containers/app/index.js [hunk 7]
```diff
@@ -19,20 +21,38 @@ class App extends Component {
...
-          <Link to="/">Home</Link>
-          <Link to="/about-us">About</Link>
-        </header>
+		<Navbar />
 
         <main>
           <Route exact path="/" component={Home} />
           <Route exact path="/about-us" component={About} />
+          <Route exact path="/navbar" component={Navbar} />
+          <Route exact path="/demo" component={Demo} />
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove line

File: client/src/containers/app/index.js [hunk 8]
```diff
@@ -19,20 +21,38 @@ class App extends Component {
...
         </main>
+
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove line

File: client/src/containers/app/index.js [hunk 9]
```diff
@@ -1,7 +1,9 @@
 import React, { Component } from 'react';
-import { Route, Link } from 'react-router-dom'
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove Link from import as it's not being used anymore

File: client/src/containers/home/index.js [hunk 10]
```diff
@@ -3,21 +3,24 @@ import { push } from 'react-router-redux'
...
       <button onClick={() => changePage()}>Go to about page via redux</button>
     </div>
   )
 }
 
 const mapStateToProps = state => {
   return {
-    message: state.app.example.message
+    //message: state.app.example.message
+    message: state.navbar
```
Comment (iandusenbury) [inline client/src/containers/home/index.js]: revert this back to original state, just so we don't have unnecessary file changes

File: client/src/containers/navbar/index.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+	[Hamburger Menu] (if signed in)
+	"View Profile"
+	"Edit Profile"
+	"View Admin"	(if admin)
+	"Sign Out"
+	"About"
+*/
+export default () => (
+	<div>
```
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: I don't think we need this wrapping div

File: client/src/containers/navbar/index.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+		<header>
+			<Toolbar>
+				<ToolbarGroup>
+					<FlatButton 
+						href="/"
+						label="MESA"
+						/>
```
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: add this to the line above
like:
```
<FlatButton
  href="/"
  label="MESA" />
```

File: client/src/containers/app/index.js (hunk 7 above)
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove line

File: client/src/containers/app/index.js (hunk 8 above)
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove line

File: client/src/containers/app/index.js [hunk 13]
```diff
@@ -19,20 +21,38 @@ class App extends Component {
...
+
+        <main>
+          <Route exact path="/" component={Home} />
+          <Route exact path="/about-us" component={About} />
+          <Route exact path="/navbar" component={Navbar} />
+          <Route exact path="/demo" component={Demo} />
+        </main>
+
+*/
+
```
Comment (iandusenbury) [inline client/src/containers/app/index.js]: delete this comment block

File: client/src/containers/home/index.js (hunk 1 above)
Comment (iandusenbury) [inline client/src/containers/home/index.js]: remove line

File: client/src/containers/home/index.js [hunk 14]
```diff
@@ -3,21 +3,24 @@ import { push } from 'react-router-redux'
...
 
+
 const Home = props => {
   const { changePage, message } = props
 
   return (
     <div>
       <h1>Home</h1>
       <p>{message}</p>
+
```
Comment (iandusenbury) [inline client/src/containers/home/index.js]: remove line

File: client/src/containers/navbar/index.js [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+						<MenuItem href="" value="admin_page" primaryText="Admin Page" />
+						<Divider />
+						<MenuItem href="" value="sign_out" primaryText="Sign Out" />
+						<MenuItem href="/about-us" value="about" primaryText="About" />
+					</IconMenu>
+				</ToolbarGroup>
+			</Toolbar>
+		</header>
+	</div>
+);
```
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: remove ';'

```
<MenuItem
              containerElement={<Link to="/about" />}
              value="about"
              primaryText="About" />
```
For the links to work correctly we need to wrap them in our router Link component. If we don't do this then the server makes a roundtrip and reloads the page with the new route. With this in place the browser doesn't need to refresh see: https://github.com/mui-org/material-ui/issues/204

File: client/package.json (hunk 6 above)
Comment (iandusenbury) [inline client/package.json]: remove anything eject and react-toolbox related
afterwards be sure to run 'yarn' on the command line to update the yarn.lock file

File: client/src/containers/app/index.js (hunk 5 above)
Comment (iandusenbury) [inline client/src/containers/app/index.js]: remove line

**Review by iandusenbury — CHANGES_REQUESTED** [review]

File: client/src/containers/app/index.js (hunk 5 above)
Comment (djarredon) [reply client/src/containers/app/index.js]: Is this supposed to include deleting "import Navbar ..." ?

File: client/src/containers/navbar/index.js [hunk 16]
```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { Route, Link } from 'react-router-dom'
+import About from '../about'
```
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: remove line, unnecessary import

File: client/src/containers/navbar/index.js [hunk 17]
```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { Route, Link } from 'react-router-dom'
```
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: remove route from import, it isn't being used

**Review by iandusenbury — CHANGES_REQUESTED** [review]

File: client/src/containers/navbar/index.js [hunk 18]
```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+//  material-ui
+import IconMenu from 'material-ui/IconMenu'
+import IconButton from 'material-ui/IconButton'
+import MenuItem from 'material-ui/MenuItem'
+import RaisedButton from 'material-ui/RaisedButton'
+import FlatButton from 'material-ui/FlatButton'
+import Divider from 'material-ui/Divider'
+import TextField from 'material-ui/TextField'
```
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: You can replace all those import statements with this:
```
import {
  IconMenu,
  IconButton,
  MenuItem,
  RaisedButton,
  FlatButton,
  Divider,
  TextField
} from 'material-ui'
```

**Review by iandusenbury — CHANGES_REQUESTED** [review]

**Review by iandusenbury — APPROVED** [review]

**Pull request merged**